src/bpftrace.cpp

```diff
@@ -118,11 +118,13 @@ int BPFtrace::add_probe(ast::ASTContext &ctx,
   auto probe = generate_probe(ap, p, usdt_location_idx);
 
   // Add the new probe(s) to resources
-  if (ap.provider == "BEGIN" || ap.provider == "END" || ap.provider == "self") {
+  if (ap.provider == "BEGIN" || ap.provider == "END") {
     // special probes
     auto target = ap.target.empty() ? "" : "_" + ap.target;
     auto name = ap.provider + target;
     resources.special_probes[name] = std::move(probe);
+  } else if (ap.provider == "self") {
+    resources.signal_probes.emplace_back(std::move(probe));
```

Comment:
For clarity, I would prefer
```suggestion
  } else if (ap.provider == "self") {
    if (ap.target == "signal")
      resources.signal_probes.emplace_back(std::move(probe));
```
or at least leave a comment that currently, `self` only supports signal probes.

Reply:
Good call. Will update.